### DISCUSSION TITLE
Code editing

### DISCUSSION THREAD

**Discussion by insberr** [community/community#3164] [Mobile]
It would be cool if you could edit code within the app.

**Comment by nplasterer** [comment]
Thanks for your feedback! We are tracking this request internally and are looking into how we can bring this to the app.

**Reply by nemkin** [reply]
I would use this for markdown files! I take notes in a Github repository with md files and I would love to be able to do it on my phone too in the native app. (There are a few third party Android apps that can do this already, for example GitJournal.io.)

**Reply by Lawrencesenin** [reply]
https://github.com/community/community/discussions/3164#discussioncomment-665975

**Comment by njradford** [comment]
Cross posting - https://github.com/github/feedback/discussions/3820#discussioncomment-676172

>  I think both a repo in general and PRs as well. it'd be useful to have an edit button present whenever a file is open. And on submit the user could be given a choice of "commit to master" or "create branch" as the Desktop UI does. I think at the moment I tend to only open the web UI on mobile if I need to make a small README change or fix a glaring bug.

**Reply by Sabaqsmar** [reply]
I agree! Having an edit button directly accessible within the repository or pull request view would greatly improve the mobile user experience. It's akin to the seamless editing capabilities we find in apps like [www.alightmotionpro.com](https://myalightmotionapk.com/), enhancing efficiency and convenience for making quick changes or fixes. This could definitely encourage more mobile usage for minor edits or bug fixes without compromising productivity.

**Comment by benhatsor** [comment]
A related project is [codeit.codes](https://codeit.codes), from which you can edit and run GitHub files on mobile.

**Comment by mohamedzhioua** [comment]
Choose [a Git client from the Play store](https://play.google.com/store/search?q=git%20client&c=apps) and install it. There are a few free options and a few paid ones. Some are essentially read-only cients, but others will let you make changes.

**Reply by Cwray9090** [reply]
Hello.....

**Reply by mohamedzhioua** [reply]
hello @Cwray9090

**Comment by mohamedzhioua** [comment]
you can visit the link below t ofind more interesting answers
https://stackoverflow.com/questions/36232183/can-i-commit-files-to-github-from-android-phone

**Comment by Wladefant** [comment]
Duplicate of #6835

**Reply by insberr** [reply]
If you actually look at the publish dates, that one is a duplicate of mine

**Reply by Wladefant** [reply]
Haha yes correct

**Comment by insberr** [comment]
The feature has been added. Would be nice if you could create/remove files using the app too though.

**Comment by alikhan-tech** [comment]
The functionality has been included. However, it would be great if you could add and remove files from the app like Alight Motion Mod APK as well.

**Comment by BlueSky-2024** [comment]
When dealing with code editing issues like #3164, it's essential to ensure that your development environment is correctly set up and that any dependencies are up to date. For quick adjustments or visual edits, using a versatile tool like [picsartapkm](https://picsartapkm.com/) can be helpful, especially if you need to create or modify images for your project.
